core/meta/src/TClass.cxx

```diff
@@ -3117,11 +3121,32 @@ TClass *TClass::GetClass(const char *name, Bool_t load, Bool_t silent, size_t hi
    // TClass if we have one.
    if (cl) return cl;
 
-   if (ispair &&  hint_pair_offset && hint_pair_size) {
-      auto pairinfo = TVirtualStreamerInfo::Factory()->GenerateInfoForPair(normalizedName, silent, hint_pair_offset, hint_pair_size);
-      //return pairinfo ? pairinfo->GetClass() : nullptr;
-      if (pairinfo)
-         return pairinfo->GetClass();
+   if (ispair) {
+      if (hint_pair_offset && hint_pair_size) {
+         auto pairinfo = TVirtualStreamerInfo::Factory()->GenerateInfoForPair(normalizedName, silent, hint_pair_offset, hint_pair_size);
+         //return pairinfo ? pairinfo->GetClass() : nullptr;
+         if (pairinfo)
+            return pairinfo->GetClass();
+      } else {
```

Comment:
Yes, if despite having the hints we can not generate it, we should fall-through to trying to generate it based on the interpreter info (added the git log for the commit that introduced this as comment).
Note that the else statement in line 3131+ is inappropriate to run here (because it will eventually call back to this routine passing some hints and if the `GenerateInfoForPair` fails, this would lead to an infinite recursion).